src/main/java/org/apache/mesos/scheduler/recovery/RecoveryRequirement.java

```diff
@@ -0,0 +1,24 @@
+package org.apache.mesos.scheduler.recovery;
+
+import org.apache.mesos.offer.OfferRequirement;
+
+/**
+ * Implementation of this interface allows RecoverySchedulers to make decisions about the scheduling of recovery of
+ * failed Tasks.
+ */
+public interface RecoveryRequirement {
+    /**
+     * Possible recovery types are None, Transient and Permanent.  None indicates that no recovery is under way.
+     * Transient indicates that non-destructive recovery is being attempted.  Permanent indicates that a destructive
+     * recovery operation is desired.
+     */
+    enum RecoveryType {
+        None,
```

Comment:
UNDERSCORE_CAPS for enums?

(or if the rest of commons is CamelCase then stick with consistency imo)